File: output/ChatGPT/codenet/Python/C++/s790754864.cpp
#include <iostream>
using namespace std;

const int mod = 1000000007;

int fact(int n, int r) {
    int result = 1;
    for (int i = 0; i < r; i++) {
        result *= (n - i);
        result /= (i + 1);
    }
    return result;
}

int main() {
    int N, K;
    cin >> N >> K;
    int R = N - K;
    
    for (int i = 1; i <= K; i++) {
        int ans;
        if (N - K + 1 < i) {
            ans = 0;
        } else {
            ans = fact(R + 1, i) * fact(K - 1, i - 1);
        }
        cout << ans % mod << endl;
    }
    
    return 0;
}

File: output/CodeLlama/codenet/Python/Java/s790754864.java
import java.util.Scanner;
import java.math.BigInteger;

public class s790754864 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();
        int R = N - K;
        BigInteger mod = new BigInteger("1000000007");
        for (int i = 1; i <= K; i++) {
            if (N - K + 1 < i) {
                System.out.println(0);
            } else {
                BigInteger ans = fact(R + 1, i).multiply(fact(K - 1, i - 1));
                System.out.println(ans.mod(mod).intValue());
            }
        }
    }

    public static BigInteger fact(int n, int r) {
        BigInteger ans = new BigInteger("1");
        for (int i = 1; i <= r; i++) {
            ans = ans.multiply(new BigInteger(String.valueOf(n - r + i)));
        }
        return ans;
    }
}